west.yml: Update hal_stm32 to fix I2C_SPEED_FAST redefinition

erwango opened the pull request:
On some STM32 series, I2C HAL defines I2C_SPEED_STANDARD and
I2C_SPEED_FAST. These definitions conflict with Zephyr I2C API.

Since Zephyr I2C STM32 driver uses LL API, we can disable I2C HAL.
Disable I2C HAL for L1 and F2 series. Deactivation is already done
2 other impacted series F1 and F4.

Additionally, on F1 series, add the commented definition line to
make the change more visible.

Requires https://github.com/zephyrproject-rtos/hal_stm32/pull/16

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

galak commented on west.yml:
```diff
@@ -51,7 +51,7 @@ manifest:
       path: modules/hal/st
     - name: hal_stm32
-      revision: 8066e1d2477860a106e3c2e6ca841b4abbfbaad2
+      revision: pull/16/head/
```
hmm, is this right?

erwango replied:
Removing the last `/`

galak commented:
@mbolivar would appreciate your comment on if we want west.yml to have explicit SHAs or the notation @erwango is using.

nashif commented:
> @mbolivar would appreciate your comment on if we want west.yml to have explicit SHAs or the notation @erwango is using.

this is documented here: https://docs.zephyrproject.org/latest/guides/modules.html

Also, you should not merge the module before the CI has completed, what is the point of CI if we merge ahead of testing? Now that the PR is merged in the module, you can probably use the SHA directly.

erwango commented:
> Also, you should not merge the module before the CI has completed, what is the point of CI if we merge ahead of testing? Now that the PR is merged in the module, you can probably use the SHA directly.

@nashif We may get Zephyr bot add a `DNM` when it detects changes to yaml with a small message on status. Plus maybe do the "back to the past" check: https://github.com/zephyrproject-rtos/zephyr/issues/16961

psidhu commented on west.yml:
```diff
@@ -51,7 +51,7 @@ manifest:
       path: modules/hal/st
     - name: hal_stm32
-      revision: 8066e1d2477860a106e3c2e6ca841b4abbfbaad2
+      revision: pull/16/head
```
Shouldn't this be a08f60768042be9b017796e13c5bb7f977d51a92?

You're stating that you want to use a PR as the HEAD for the STM32 HAL code.

erwango replied:
@psidhu , yeah, I'm actually following the process documented here: https://docs.zephyrproject.org/latest/guides/modules.html?highlight=modules#process
Now that CI has been run successfully, `pull/16/head` has been merged and  I can change to SHA1 of the merged commit.

ioannisg commented:
> > @mbolivar would appreciate your comment on if we want west.yml to have explicit SHAs or the notation @erwango is using.
> 
> this is documented here: https://docs.zephyrproject.org/latest/guides/modules.html
> 
> Also, you should not merge the module before the CI has completed, what is the point of CI if we merge ahead of testing? Now that the PR is merged in the module, you can probably use the SHA directly.

I agree, this should be the right approach.

erwango committed:
west.yml: Update hal_stm32 to fix I2C_SPEED_FAST redefinition

On some STM32 series, I2C HAL defines I2C_SPEED_STANDARD and
I2C_SPEED_FAST. These definitions conflict with Zephyr I2C API.

Since Zephyr I2C STM32 driver uses LL API, we can disable I2C HAL.
Disable I2C HAL for L1 and F2 series. Deactivation is already done
2 other impacted series F1 and F4.

Additionally, on F1 series, add the commented definition line to
make the change more visible.

Requires https://github.com/zephyrproject-rtos/hal_stm32/pull/16

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>
